### DISCUSSION TITLE
Control / Filter the (sudo) commands for specific user or group

### DISCUSSION THREAD

**Discussion by gowthamsadasivam** [gravitational/teleport#7599] [Q&A]
Hello everyone,

I've started to experiment with teleport very recently for centralised SSH access management comparing with FreeIPA. I really liked the way how Teleport can record and replay the SSH commands ran by any user.

I would like to know is there a way to whitelist or blacklist a set of (sudo) commands for a specific user/group. For example the developer group member cannot run a command `(sudo systemctl disable firewalld)` that will disable firewall, but a ops group member can do that.

In FreeIPA this can be achieved by setting up sudo rules for specific user or group. I can see [Teleport can be configurable with PAM](https://goteleport.com/docs/server-access/guides/ssh-pam/) but I don't see any documentation on how to dynamically modify sudo rules for a specific unix user or group.

Would like to know if this is possible to achieve using Teleport. May not be a fully automated or out of the box feature but any manual way is also fine.

**Comment by wadells** [comment]
Hi @gowthamsadasivam,

We wrote in depth about this feature here: https://goteleport.com/blog/ssh-restricted-shells/
If you're not already familiar with how restricting shells can fail, I'd read that first.

I'd recommend you investigate setting up different unix users for the different groups, and using filesystem permissions to grant limited access to the binaries/files that specific users need.  Particularly, maybe don't allow the developer user sudo access, as that can almost always be used to disrupt system config whether a specific command is disallowed.  For example in the firewalld case, what is different about disabling firewalld vs configuring it more permissively? How many different ways can you think of to edit firewalld rules (or add other, conflicting network controls) with sudo access? Focusing on what is allowed is typically safer than focusing on what isn't allowed. 

That said, watch for our upcoming 7.0 release: https://goteleport.com/docs/preview/upcoming-releases/#teleport-70-stockholm We have a eBPF based restriction under development -- though I'm not sure it is relevant to your use case yet.  More here: https://github.com/gravitational/teleport/pull/7099
